Source/WebCore/editing/Editing.cpp

```diff
@@ -1261,29 +1266,39 @@ static std::optional<unsigned> visualDistanceOnSameLine(const RenderedPosition&
     return std::nullopt;
 }
 
-static std::optional<BoundaryPoint> visuallyClosestBidiBoundary(const RenderedPosition& position, unsigned bidiLevel)
+static std::optional<BoundaryPoint> findBidiBoundary(const RenderedPosition& position, unsigned bidiLevel, SelectionExtentMovement movement)
 {
+    auto makeBoundaryPoint = [](const RenderedPosition& position) -> std::optional<BoundaryPoint> {
```

Comment:
Maybe we can add this to RenderedPosition.h?

Reply:
Makes sense — created a new helper method on `RenderedPosition`, and used it here:

```
std::optional<BoundaryPoint> boundaryPoint() const;
```